syntax = "proto3";

message KeyValuePair {
   int32 key = 1;
   string value = 2;
}

message InsertionStatus {
   int32 status = 1;
}

message KeyToBeChecked {
   int32 key = 1;
}

message ValueChecked {
   string value = 1;
}

message ServiceIdentifier {
   string serviceIdentifier = 1;
}

message ServiceStatus {
   int32 status = 1;
}

message EndExecutionStatus {
   int32 status = 1;
}

message NoParameterKeyValue {}

service KeyValue {
   rpc InsertKeyValue (KeyValuePair) returns (InsertionStatus) {}
   rpc CheckoutValue (KeyToBeChecked) returns (ValueChecked) {}
   rpc Activate (ServiceIdentifier) returns (ServiceStatus) {}
   rpc EndExecution (NoParameterKeyValue) returns (EndExecutionStatus) {}
}